pkg/userdata/centos/testdata/kubelet-v1.10-aws.golden

```diff
@@ -244,11 +260,95 @@ write_files:
     After=docker.service
     
     [Service]
+    Environment="PATH=/opt/bin:/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin/"
     ExecStart=/opt/bin/health-monitor.sh container-runtime
     
     [Install]
     WantedBy=multi-user.target
     
 
+- path: /etc/systemd/system/docker.service
+  permissions: "0644"
+  content: |
+    [Unit]
+    Description=Docker Application Container Engine
+    Documentation=https://docs.docker.com
+    After=containerd.service docker.socket network-online.target
+    Wants=network-online.target
+    Requires=containerd.service docker.socket
+    
+    [Service]
+    Environment="PATH=/opt/bin:/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin/"
+    Type=notify
+    # the default is not to use systemd for cgroups because the delegate issues still
+    # exists and systemd currently does not support the cgroup feature set required
+    # for containers run by docker
+    ExecStart=/opt/bin/dockerd -H fd:// --containerd=/var/run/docker/libcontainerd/docker-containerd.sock 
+    ExecReload=/bin/kill -s HUP $MAINPID
+    LimitNOFILE=1048576
+    # Having non-zero Limit*s causes performance problems due to accounting overhead
+    # in the kernel. We recommend using cgroups to do container-local accounting.
+    LimitNPROC=infinity
+    LimitCORE=infinity
+    # Uncomment TasksMax if your systemd version supports it.
+    # Only systemd 226 and above support this version.
+    
+    TimeoutStartSec=0
+    # set delegate yes so that systemd does not reset the cgroups of docker containers
+    Delegate=yes
+    # kill only the docker process, not all processes in the cgroup
+    KillMode=process
+    # restart the docker process if it exits prematurely
+    Restart=on-failure
+    StartLimitBurst=3
+    StartLimitInterval=60s
+    
+    [Install]
+    WantedBy=multi-user.target
+
+- path: /etc/systemd/system/docker.socket
```

Comment:
Is there any reason to have a `docker.socket` and a `containerd.service` aside from "CoreOS does it that way"? The docker package on Ubuntu manages perfectly fine without a `containerd.service` and the socket is only useful to have socket activation which we don't need because we already explicitly declare `docker` as dependency of the Kubelet

Reply:
`containerd.service`:
Even though has a compiled in containerd, docker will look on the host for a `/usr/bin/containerd` first. If one is found, that binary will be started instead of the compiled in one.
On CoreOS `/usr/bin/containerd` exists - as it gets shipped with CoreOS, but does not work together with Docker 17.04. As Thus the way via the service is needed.

`docker.socket`
That is how CoreOS does it. I can remove it

Reply:
Done with https://github.com/kubermatic/machine-controller/pull/382/commits/ce9322d27ec5c9706903b536d7c67294a104cdde

Reply:
> containerd.service:
Even though has a compiled in containerd, docker will look on the host for a /usr/bin/containerd first. If one is found, that binary will be started instead of the compiled in one.

Yeah but that would mean that we need it on coreos only, not on the rest. I'd really like to prefer all of this as simple as possible and right now IMHO the templates are extremely hard to read already, so I'd prefer to not put anything in them which we dont absolutely need

Reply:
We can use a drop-in to add the additional dependency without having to alter the main unit file

Reply:
Great idea, i'll adapt the pull!